pyccel/decorators.py

```diff
@@ -130,6 +130,40 @@ def kernel(f):
     KernelAccessor
         A class representing the kernel function.
     """
+    class CudaThreadIndexing:
+        """
+        Class representing the CUDA thread indexing.
+
+        Class representing the CUDA thread indexing.
+        """
```

Comment:
The classes inside this `kernel` function will be recreated at each kernel call. I think it would make more sense to put them somewhere in pyccel/cuda

Reply:
528099f is that better